torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -271,6 +285,42 @@ static void UpdateScalarTypeForOutput(
       CreateProfiledTensorTypeWithScalarType(output_tensor_type, scalar_type));
 }
 
+static void RecoverScalarTypeForOutput(
+    Node* n,
```

Comment:
safer to pass in `Value* out` instead of `Node* n`, because later `n->output()` is called to retrieve the output, and that call assumes the node `n` has a single output, which will raise runtime error if that's not the case.

Reply:
On the other hand, you can get `n` back by calling `out->node()`, although need to do `TORCH_INTERNAL_ASSERT(nullptr != n)` to be safe.

Reply:
Thanks, has changed it to Value* out.